File: src/main/java/github/buried_in_code/pieces/Archer.java
package github.buried_in_code.pieces;

import github.buried_in_code.Direction;
import github.buried_in_code.players.Player;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Created by Macro303 on 2018-Feb-08.
 */
public class Archer extends Piece {
	public Archer(@NotNull Player player) {
		super(player, 6, "A", List.of(
				Direction.NORTH,
				Direction.EAST,
				Direction.SOUTH,
				Direction.WEST
		));
	}

	@Override
	public String toString() {
		return "Archer{} " + super.toString();
	}
}

File: Base/src/main/kotlin/macro303/base/pieces/Archer.kt
package macro303.base.pieces

import macro303.base.Colour
import macro303.base.Direction

/**
 * Created by Macro303 on 2017-10-20.
 */
class Archer(teamColour: Colour) : Piece(
	teamColour = teamColour,
	maxDistance = 6,
	validDirections = arrayOf(Direction.NORTH, Direction.EAST, Direction.SOUTH, Direction.WEST)
) {
	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (other !is Archer) return false
		if (!super.equals(other)) return false
		return true
	}

	override fun hashCode(): Int {
		return super.hashCode()
	}

	override fun toString(): String {
		return "Archer() ${super.toString()}"
	}
}